index.bs

```diff
@@ -1462,3 +1484,231 @@ The [=remote end steps=] with |command parameters| are:
+          contexts: [*BrowsingContextInfo]
+        }
+    </pre>
+   </dd>
+</dl>
+
+<div algorithm="remote end steps for browsingContext.list">
+The [=remote end steps=] with |command parameters| are:
+
+  1. Let the |parent id| be the value of the <code>parent</code> field of
+       |command parameters| if present, or null otherwise.
+
+  1. Let |parent| be the result of [=trying=] to [=get a browsing context=]
+     given |parent id|.
+
+  1. If |parent| is null, let |contexts| be a list containing all [=top-level
+     browsing contexts=]. Otherwise let |contexts| be a list containing all
+     [=/browsing contexts=] which are [=child browsing contexts=] of |parent|.
+
+  1. Let |context info| be a list.
+
+  1. For each |context| of |context|, append the result of [=get the browsing
+     context info=] given |context| to |context info|.
+
+  1. Let |body| be a map matching the <code>BrowsingContextListResult</code>
+     production, with the <code>contexts</code> field set to |context info|.
+
+  1. Return [=success=] with data |body|.
+
+Issue: Should this contain other data like the current URL for the context.
+
+</div>
+
+### Events ### {#module-contexts-events}
+
+#### contextCreated #### {#event-browsingContext-contextCreated}
+
+<dl>
+   <dt>Event Type</dt>
+   <dd>
+      <pre class="cddl local-cddl">
+        BrowsingContextContextCreatedEvent = {
+         method: "browsingContext.contextCreated",
+         params: BrowsingContextInfo
+       }
+      </pre>
+   </dd>
+</dl>
+The [=remote end event trigger=] is:
+
+<div algorithm="remote end event trigger for browsingContext.contextCreated">
+When the [=create a new browsing context=] algorithm is invoked, after the
+[=active document=] of the browsing context is set, run the following steps:
+
+ 1. Let |context| be the newly created browsing context.
+
+ 1. Let |related contexts| be a set containing the [=parent browsing context=]
+    of |context|, if that is not null, or an empty set otherwise.
+
+ 1. Let |params| be the result of [=get the browsing context info=] given
+    |context|.
+
+ 1. Let |body| be a map matching the
+    <code>BrowsingContextContextCreatedEvent</code> production, with the
+    <code>value</code> field set to |params|.
+
+ 1. [=Emit an event=] with |body| and |related contexts|.
+
+</div>
+
+#### contextDestroyed #### {#event-browsingContext-contextDestroyed}
+
+<dl>
+   <dt>Event Type</dt>
+   <dd>
+      <pre class="cddl local-cddl">
+        BrowsingContextContextDestroyedEvent = {
+         method: "browsingContext.contextDestroyed",
+         params: BrowsingContextDestroyedParameters
+       }
+
+       BrowsingContextDestroyedParameters = {
+         context: BrowsingContext,
+         parent: BrowsingContext / null,
+       }
+      </pre>
+   </dd>
+</dl>
+The [=remote end event trigger=] is:
+
+<div algorithm="remote end event trigger for browsingContext.contexDestroyed">
+
+When [=a browsing context is discarded=], run the following steps:
```

Comment:
https://html.spec.whatwg.org/#a-browsing-context-is-discarded has a funny ID, but like "create a new browsing context" it's an algorithm that's invoked, not an event that happens. So I think this needs to be "When the discard a browsing context algorithm is invoked". Some creative linking may be needed.